File: src/main/java/org/taktik/icure/entities/embed/Employer.java
package org.taktik.icure.entities.embed;

import java.io.Serializable;
import java.util.Objects;

public class Employer implements Serializable {
    private String name;
    private Address addresse;

    public String getName() { return name; }

    public void setName(String name) { this.name = name; }

    public Address getAddresse() { return addresse; }

    public void setAddresse(Address addresse) { this.addresse = addresse; }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Employer employer = (Employer) o;
        return Objects.equals(name, employer.name) &&
                Objects.equals(addresse, employer.addresse);
    }

    @Override
    public int hashCode() {
        return Objects.hash(name, addresse);
    }
}

File: src/main/kotlin/org/taktik/icure/entities/embed/Employer.kt
package org.taktik.icure.entities.embed

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonInclude
import com.github.pozo.KotlinBuilder
import org.taktik.icure.entities.base.Named
import java.io.Serializable

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
@KotlinBuilder
data class Employer(
        override val name: String? = null,
        val addresse: Address? = null
) : Named, Serializable
